src/overlays/actors/ovl_Bg_Hidan_Dalm/z_bg_hidan_dalm.c

```diff
@@ -21,17 +24,176 @@ const ActorInit Bg_Hidan_Dalm_InitVars = {
     (ActorFunc)BgHidanDalm_Update,
     (ActorFunc)BgHidanDalm_Draw,
 };
-*/
+
+static ColliderTrisItemInit sTrisItemInit[4] = {
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000040, 0x00, 0x00 }, 0x00, 0x79, 0x00 },
+        { { { 305.0f, 0.0f, -300.0f }, { 305.0f, 600.0f, -300.0f }, { 305.0f, 600.0f, 300.0f } } },
+    },
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000040, 0x00, 0x00 }, 0x00, 0x79, 0x00 },
+        { { { 305.0f, 0.0f, -300.0f }, { 305.0f, 600.0f, 300.0f }, { 305.0f, 0.0f, 300.0f } } },
+    },
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000040, 0x00, 0x00 }, 0x00, 0x79, 0x00 },
+        { { { -305.0f, 0.0f, -300.0f }, { -305.0f, 600.0f, 300.0f }, { -305.0f, 600.0f, -300.0f } } },
+    },
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000040, 0x00, 0x00 }, 0x00, 0x79, 0x00 },
+        { { { -305.0f, 0.0f, -300.0f }, { -305.0f, 0.0f, 300.0f }, { -305.0f, 600.0f, 300.0f } } },
+    },
+};
+
+ColliderTrisInit D_80886590 = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x00, 0x20, COLSHAPE_TRIS },
+    4,
+    sTrisItemInit,
+};
+
+InitChainEntry D_808865A0[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(gravity, 65336, ICHAIN_STOP),
+};
+
+extern Gfx D_0600BBF0[];
+extern Gfx D_0600BDF0[];
+extern UNK_TYPE D_0600DA10;
+
+#ifdef NON_MATCHING
+// regalloc
+void BgHidanDalm_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgHidanDalm* this = THIS;
+    s32 switchFlag;
+    u32 dynaUnk;
+
+    dynaUnk = 0;
+    Actor_ProcessInitChain(thisx, D_808865A0);
+    DynaPolyInfo_SetActorMove(&this->dyna, DPM_UNK);
+    DynaPolyInfo_Alloc(&D_0600DA10, &dynaUnk);
+    // = ... & 0xFFFFFFFFFFFFFFFF; moves regalloc issues
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, dynaUnk);
+    Collider_InitTris(globalCtx, &this->unk_16C);
+    Collider_SetTris(globalCtx, &this->unk_16C, thisx, &D_80886590, this->unk_18C);
+
+    switchFlag = thisx->params >> 8; /*
+    // this instead of the line above solves half the regalloc issues
+    switchFlag = thisx->params;
+    switchFlag = switchFlag >> 8 & 0xFFFFFFFFFFFFFFFF;
+    //*/
+    this->switchFlag = switchFlag;   // can use thisx->params >> 8 without changes
+    thisx->params &= 0xFF;
+    // can use thisx->params >> 8 , seems to improve regalloc, but it changes codegen
+    if (Flags_GetSwitch(globalCtx, switchFlag & 0xFF)) {
```

Comment:
This should match:
```suggestion
    this->switchFlag = (thisx->params >> 8) & 0xFF;
    thisx->params &= 0xFF;

    if (Flags_GetSwitch(globalCtx, this->switchFlag)) {
```

For the explanation, the `& 0xFF` in the `Flags_GetSwitch` call is just an implicit `u8` cast. As a general rule, when something is read after it was just written, the compiler often optimizes out the reload, but it casts the value so it's actually equivalent to a load. So here, after writing inside `this->switchFlag`, it keeps the register and casts it to `u8` since that's the type of `this->switchFlag`, which looks like a variable but it's not.

As for the actual `& 0xFF` when writing `this->switchFlag`, it's optimized out since it stores it as a byte anyway, but it's important for regalloc.

And you can replace the `switchFlag` temp with a pad since that should still be needed for stack allocation.